src/ui.form-date-picker-range/composables/attrs.composable.js

```diff
@@ -7,124 +7,136 @@ import defaultConfig from "../configs/default.config";
 import { POSITION } from "../../composable.adjustElementPosition";
 
 export default function useAttrs(props, { isShownMenu, isTop, isRight }) {
-  const { config, getAttrs } = useUI(defaultConfig, () => props.config);
-  const { menu } = config.value;
+  const { config, getAttrs, isSystemKey } = useUI(defaultConfig, () => props.config);
+  const attrs = {};
 
   const openDirectionY = computed(() => (isTop.value ? POSITION.top : POSITION.bottom));
   const openDirectionX = computed(() => (isRight.value ? POSITION.right : POSITION.left));
 
-  const cvaMenu = cva({
-    base: menu.base,
-    variants: menu.variants,
-    compoundVariants: menu.compoundVariants,
-  });
-
   const menuClasses = computed(() =>
-    cvaMenu({ openDirectionY: openDirectionY.value, openDirectionX: openDirectionX.value }),
+    cva(config.value.menu)({
+      openDirectionY: openDirectionY.value,
+      openDirectionX: openDirectionX.value,
+    }),
   );
 
-  const wrapperAttrs = getAttrs("wrapper");
-  const buttonWrapperAttrsRaw = getAttrs("buttonWrapper");
-  const buttonAttrsRaw = getAttrs("button");
-  const shiftRangeButtonAttrs = getAttrs("shiftRangeButton");
-  const calendarAttrs = getAttrs("calendar", { isComponent: true });
-  const inputBlurAttrs = getAttrs("input", { isComponent: true });
-  const inputActiveAttrs = getAttrs("inputActive");
-  const rangeInputAttrs = getAttrs("rangeInput", { isComponent: true });
-  const rangeInputWrapperAttrs = getAttrs("rangeInputWrapper");
-  const inputRangeErrorAttrs = getAttrs("inputRangeError");
-
-  // This watcher rewrites default calendar locales with datepicker range locales
-  // Watcher will not rewrite custom calendar locales
-  watchEffect(() => {
-    if (!calendarAttrs.value.config) {
-      calendarAttrs.value.config = {};
+  for (const key in defaultConfig) {
+    if (isSystemKey(key)) continue;
+
+    const classes = computed(() => {
+      const value = config.value[key];
+
+      if (value.variants || value.compoundVariants) {
+        return cva(value)({
+          ...props,
+        });
+      }
+
+      return "";
+    });
+
+    attrs[`${key}Attrs`] = getAttrs(key, { classes });
+
+    if (key === "menu") {
+      const menuAttrs = attrs[`${key}Attrs`];
+
+      attrs[`${key}Attrs`] = computed(() => ({
+        ...menuAttrs.value,
+        class: cx([menuClasses.value, menuAttrs.value.class]),
```

Comment:
this if condition no needed at all

Reply:
I left this if condition in latest commit as it's the only place where these constants with props **"isTop"/"isRight"** are used:
```
const openDirectionY = computed(() => (isTop.value ? POSITION.top : POSITION.bottom));
const openDirectionX = computed(() => (isRight.value ? POSITION.right : POSITION.left));
```